apps/api/lib/viewCollection.js

```diff
@@ -0,0 +1,26 @@
+import { sparql } from '@tpluscode/sparql-builder'
+import { schema, vcard } from '@tpluscode/rdf-ns-builders'
+
+export function textFilter({ subject, object, variable }) {
+  if (!object.value) {
+    return ''
+  }
+
+  const q = variable('q')
+
+  return sparql`
+    {
+      ${subject} ${schema.name} ${q}
+    }
+    UNION
+    {
+      ${subject} ${schema.alternateName} ${q}
+    }
+    UNION
+    {
+      ${subject} ${schema.author}/${vcard.hasUID} ${q}
+    }
+  
+    FILTER( REGEX(str(${q}), "${object.value}", "i") )
+  `
+}
```

Comment:
Collection filter is describe on https://creta.hypermedia.app/#/knossos/collections?id=queries

Reply:
How about establishing a minimum character limit for auto-complete/filter? say 2-4 chars?

Two main reasons:
	- The performance, there is no need to trigger the call if the results are huge
	- The user's cognitive load, the results should be directly helpful to the user, otherwise might be a distraction.
	
	It's what I've seen in the wild.
	
	Perhaps it can be added as a default parameter.

Reply:
Yes, good point. This is already part of the form support for queries. See note under https://forms.hypermedia.app/#/extensions/hydra?id=collection-filtered-with-text-input

If absent, the default is `1`. Just added that to these docs

Reply:
Ah, this is done slightly different here. Updated